### PR TITLE
Cc 0012 front end cc galaxy project init

### PR DESCRIPTION
**Pull request by VanderDeng**
init the frontend project:

- Add dependencies

- README file update

- Axios setting

- Style&theme structure

- Splitting of page structure components

- Complete Tests

### CODE REVIEW

File: src/utils/axios.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import axios from "axios";
+import _ from "lodash";
```
Comment (tatoMa) [inline src/utils/axios.ts:2]: Do you think we need the whole lodash lib as you import it all. 
Just asking.

Comment (VanderDeng) [reply src/utils/axios.ts:2]: I don't think we need to import the whole lib.
How about I change it to only import the necessary tool from 'lodash-es'

Comment (tatoMa) [reply src/utils/axios.ts:2]: That's would be way better.

File: src/pages/index.tsx [hunk 2]
```diff
@@ -1,109 +1,15 @@
...
-                >
-                  Starting at $0.01/mo
-                </Text>
-              </Box>
-            </Stack>
-          </Stack>
-        </Container>
-      </Flex>
+    <HeaderLayout>
+      <Heading as="h1" size="xl" paddingLeft="326px">
```
Comment (tatoMa) [inline src/pages/index.tsx:8]: paddingLeft="326px"?
I know it is just a placeholder but it doesn't make any sense to make it padding left at 326px.

File: src/store/reducer/counterSlice.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+export const incrementIfOdd =
+  (amount: number): AppThunk =>
+  (dispatch, getState) => {
+    const currentValue = selectCount(getState());
+    if (currentValue % 2 === 1) {
+      dispatch(incrementByAmount(amount));
+    }
+  };
+
+export default counterSlice.reducer;
```
Comment (tatoMa) [inline src/store/reducer/counterSlice.ts:42]: What is this counter slice for? At least have it on the index page or delete it.

File: src/constants/environment.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const environment = {
+  apiBaseUrl: process.env.NEXT_PUBLIC_API_BASE_URI,
+};
```
Comment (tatoMa) [inline src/constants/environment.ts:3]: Uppercase for constant identifiers?

File: src/layouts/index.tsx [hunk 5]
```diff
@@ -11,9 +10,7 @@ const Layout: React.FC<LayoutProps> = ({ children }) => {
   return (
     <>
       <Flex as="header" width="full" align="center">
-        <Box marginLeft="auto" marginTop={5} marginRight={5}>
-          <ThemeToggle />
-        </Box>
+        <Box marginLeft="auto" marginTop={5} marginRight={5}></Box>
```
Comment (tatoMa) [inline src/layouts/index.tsx:13]: Do we still need this Box and even Flex components after the theme toggle button has been removed?

File: src/pages/_document.tsx [hunk 6]
```diff
@@ -1,61 +1,33 @@
...
+    // Retrieve styles from components in the page
+    const originalRenderPage = ctx.renderPage;
+
+    try {
+      ctx.renderPage = () =>
+        originalRenderPage({
+          enhanceApp: (App) => (props) => sheet.collectStyles(<App {...props} />),
+        });
+
+      const initialProps = await Document.getInitialProps(ctx);
```
Comment (tatoMa) [inline src/pages/_document.tsx:18]: Explain why use getInitialProps next api.

Comment (tatoMa) [reply src/pages/_document.tsx:18]: Or the reason why you using SSR apis.

File: src/__tests__/layouts/HeaderLayout.test.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import Home from "../../pages";
+
+describe("Header", () => {
+  it("shoud render layout success", () => {
+    const { debug } = renderWithMockedProvider(
+      <HeaderLayout title="Home">
+        <Home />
+      </HeaderLayout>
+    );
+    debug(undefined, Infinity);
```
Comment (kritomomo) [inline src/__tests__/layouts/HeaderLayout.test.tsx:12]: Do you still need the debug here?

**Review by kritomomo — COMMENTED** [review]
For the layout test, Since we don't have something to expect, I think we can choose not to test it(just ignore the layout folder), or try https://jestjs.io/docs/snapshot-testing.

File: src/store/reducer/counterSlice.ts (hunk 3 above)
Comment (VanderDeng) [reply src/store/reducer/counterSlice.ts:42]: It is a placeholder. if it be deleted, next will report an error saying that the reducer is missing from the store

File: src/layouts/index.tsx (hunk 5 above)
Comment (VanderDeng) [reply src/layouts/index.tsx:13]: I will deleted that one

File: src/__tests__/layouts/HeaderLayout.test.tsx (hunk 7 above)
Comment (VanderDeng) [reply src/__tests__/layouts/HeaderLayout.test.tsx:12]: ok, I prefer to ignore the layout test.

File: src/pages/index.tsx (hunk 2 above)
Comment (VanderDeng) [reply src/pages/index.tsx:8]: I will fix that

File: src/pages/_document.tsx (hunk 6 above)
Comment (VanderDeng) [reply src/pages/_document.tsx:18]: My fault, I will rollback this file to the original one

File: src/constants/environment.ts (hunk 4 above)
Comment (VanderDeng) [reply src/constants/environment.ts]: I will fix that one

File: src/pages/_document.tsx (hunk 6 above)
Comment (VanderDeng) [reply src/pages/_document.tsx]: Now it is the same as the main branch

File: src/pages/index.tsx [hunk 8]
```diff
@@ -1,109 +1,15 @@
...
-                >
-                  Starting at $0.01/mo
-                </Text>
-              </Box>
-            </Stack>
-          </Stack>
-        </Container>
-      </Flex>
+    <HeaderLayout>
+      <Heading as="h1" size="xl">
```
Comment (tatoMa) [inline src/pages/index.tsx:8]: if you want to center the text, try props: textAlign="center"

**Review by tatoMa — APPROVED** [review]

File: src/pages/index.tsx (hunk 8 above)
Comment (VanderDeng) [reply src/pages/index.tsx]: Ok, I fixed that one

**Review by yuanningd — APPROVED** [review]

**Pull request merged**